File: WebSharper.Templates/templates/SPA-CSharp/Startup.cs
using WebSharper.AspNetCore;
using WebSharper.SPA.CSharp;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.
builder.Services.AddWebSharper()
                .AddAuthentication("WebSharper")
                .AddCookie("WebSharper", options => { });

var app = builder.Build();

// Configure the HTTP request pipeline.
if (!app.Environment.IsDevelopment())
{
    app.UseExceptionHandler("/Error");
    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
    app.UseHsts();
}

app.UseHttpsRedirection();

//-:cnd:noEmit
#if DEBUG        
app.UseWebSharperScriptRedirect(startVite: true);
#endif
//+:cnd:noEmit

app.UseDefaultFiles();

app.UseStaticFiles();

//Enable if you want to make RPC calls to server
//app.UseWebSharperRemoting();

app.Run();

File: WebSharper.Templates/templates/SPA-FSharp/Startup.fs
open System
open Microsoft.AspNetCore.Builder
open Microsoft.Extensions.Hosting
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.DependencyInjection
open WebSharper.AspNetCore
open WebSharper.SPA.FSharp

[<EntryPoint>]
let main args =
    let builder = WebApplication.CreateBuilder(args)
    
    // Add services to the container.
    builder.Services.AddWebSharper()
        .AddAuthentication("WebSharper")
        .AddCookie("WebSharper", fun options -> ())
    |> ignore

    let app = builder.Build()

    // Configure the HTTP request pipeline.
    if not (app.Environment.IsDevelopment()) then
        app.UseExceptionHandler("/Error")
            // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
            .UseHsts()
        |> ignore
    
    app.UseHttpsRedirection()
//-:cnd:noEmit
#if DEBUG        
        .UseWebSharperScriptRedirect(startVite = true)
#endif
//+:cnd:noEmit
        .UseDefaultFiles()
        .UseStaticFiles()
        //Enable if you want to make RPC calls to server
        //.UseWebSharperRemoting()
    |> ignore 
       
    app.Run()

    0 // Exit code
